routing/pathfind.go

```diff
@@ -832,8 +835,12 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 				continue
 			}
 
+			var amtForPolicy lnwire.MilliSatoshi
+			if !splitNeeded || fromNode != source {
+				amtForPolicy = amtToSend
+			}
 			policy := unifiedPolicy.getPolicy(
-				amtToSend, g.bandwidthHints,
+				amtForPolicy, g.bandwidthHints,
```

Comment:
I am not sure if pathfinding is the best place to do a just-in-time split. How does this compare to evaluating local balances and potentially lowering the payment amount before calling pathfinding?